File: Concesionario/src/gestorAplicacion/personal/Cliente.java
package gestorAplicacion.personal;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Scanner;
import gestorAplicacion.activos.*;

public class Cliente implements Persona, Serializable{

	private static final long serialVersionUID = 1L;
	static ArrayList<Cliente> clientes = new ArrayList<Cliente>();
	protected String nombre;
	protected long cedula;
	protected long telefono;
	protected String correo;
	protected String direccion;
	private String modeloInteres;
	private long presupuesto;
	private Auto auto;

	public Cliente(String nombre, long cedula, long telefono, String correo, String direccion, String modeloInteres,
			long presupuesto) {
		this.nombre = nombre;
		this.cedula = cedula;
		this.telefono = telefono;
		this.correo = correo;
		this.direccion = direccion;
		this.modeloInteres = modeloInteres;
		this.presupuesto = presupuesto;
		Cliente.clientes.add(this);
	}

	public Cliente(String nombre, long cedula, long telefono, String correo, String modeloInteres, long presupuesto) {
		this(nombre,cedula,telefono,correo,"Medellin",modeloInteres,presupuesto);
	}

	public String getModeloInteres() {
		return modeloInteres;
	}

	public long getPresupuesto() {
		return presupuesto;
	}

	public void setModeloInteres(String modeloInteres) {
		this.modeloInteres = modeloInteres;
	}

	public void setPresupuesto(long presupuesto) {
		this.presupuesto = presupuesto;
	}
	
	public void setAuto(Auto auto) {
		this.auto = auto;
	}
	
	public Auto getAuto() {
		return auto;
	}
	
	public static ArrayList<Cliente> getClientes() {
		return clientes;
	}
	
	public static Cliente getClientePorCedula(long cedula){
		Cliente finder = null;
		for(Cliente cliente: clientes) {
			if(cliente.getCedula() == cedula) {
				finder = cliente;
				break;
			}
		}
		return finder;
	}

	public static void setClientes(ArrayList<Cliente> clientes) {
		Cliente.clientes = clientes;
	}
	@Override
	public String info() {
		String texto = "Nombre del cliente: " + getNombre() + "\n";
		return texto;
	}
	@Override
	public String getNombre() {
		return nombre;
	}

	@Override
	public long getCedula() {
		return cedula;
	}

	@Override
	public long getTelefono() {
		// TODO Auto-generated method stub
		return telefono;
	}

	@Override
	public String getCorreo() {
		// TODO Auto-generated method stub
		return correo;
	}

	@Override
	public String getDireccion() {
		// TODO Auto-generated method stub
		return direccion;
	}

	@Override
	public void setNombre(String nombre) {
		this.nombre=nombre;
	}

	@Override
	public void setCedula(long cedula) {
		this.cedula=cedula;
	}

	@Override
	public void setTelefono(long telefono) {
		this.telefono=telefono;
	}

	@Override
	public void setCorreo(String correo) {
		this.correo=correo;
	}

	@Override
	public void setDireccion(String direccion) {
		this.direccion=direccion;
	}
}

File: PYTHON/Concesionario/gestorAplicacion/Personal/cliente.py
import sys
import os
ruta_activos = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Activos'))
ruta_personal = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Personal'))
ruta_gestor = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
sys.path.append(ruta_activos)
sys.path.append(ruta_personal)
sys.path.append(ruta_gestor)


from persona import Persona

class Cliente(Persona):
    clientes = []

    def __init__(self, nombre, cedula, telefono, correo, modeloInteres, presupuesto, direccion="Medellin"):
        self.nombre = nombre
        self.cedula = cedula
        self.telefono = telefono
        self.correo = correo
        self.direccion = direccion
        self.modeloInteres = modeloInteres
        self.presupuesto = presupuesto
        Cliente.clientes.append(self)

    def get_modeloInteres(self):
        return self.modeloInteres

    def get_presupuesto(self):
        return self.presupuesto

    def set_modeloInteres(self, modeloInteres):
        self.modeloInteres = modeloInteres

    def set_presupuesto(self, presupuesto):
        self.presupuesto = presupuesto

    def set_auto(self, auto):
        self.auto = auto

    def getAuto(self):
        return self.auto

    @classmethod
    def getClientes(cls):
        return Cliente.clientes

    @staticmethod
    def getClientePorCedula(cedula):
        finder=None
        for cliente in Cliente.clientes:
            if cliente.get_cedula() == cedula:
                finder = cliente
                break
        return finder

    def set_clientes(client):
        Cliente.clientes = client

    def info(self):
        texto = "Nombre del cliente: " + self.get_nombre() + "\n"
        return texto
    
    def get_cedula(self):
        return self.cedula

    def get_correo(self):
        return self.correo

    def get_direccion(self):
        return self.direccion

    def get_nombre(self):
        return self.nombre

    def get_telefono(self):
        return self.telefono

    def set_cedula(self, cedula):
        self.cedula = cedula

    def set_correo(self, correo):
        self.correo = correo

    def set_direccion(self, direccion):
        self.direccion = direccion

    def set_nombre(self, nombre):
        self.nombre = nombre

    def set_telefono(self, telefono):
        self.telefono = telefono